### PR TITLE
Server Initial State

### PR DESCRIPTION
**Pull request by sfyll**
Silicon is an abstraction for memory, which in our case is each pool state, or to be strictly exact, both the "marketModels" and "encryptedMarketModels". As such, we use this class to abstract reads and writes away.

We then verify knowledge of all pre-images by verifying that hash(pre-image) == image for all images and pre-images, which should be of equal number.

The other commits are mostly packaging internals.

### CODE REVIEW

File: client/server/silicon/silicon.ts
Comment (lyronctk) [inline client/server/silicon/silicon.ts:7]: Great abstraction!

File: client/server/silicon/silicon.ts
Comment (lyronctk) [inline client/server/silicon/silicon.ts:22]: please add natural language comments on top of every function (if you're keeping these)

Comment (lyronctk) [reply client/server/silicon/silicon.ts:22]: otherwise, can label as 
```
/*
 * [TMP]
 */
```

File: client/server/silicon/silicon_query.ts
Comment (lyronctk) [inline client/server/silicon/silicon_query.ts:2]: big fan of this dir structure!

File: client/server/silicon/silicon_utils.ts
Comment (lyronctk) [inline client/server/silicon/silicon_utils.ts:13]: We usually put our entire shielded structure into one hash instead of having each property be a hash because it's much fewer constraints to do a poseidon hash of size 2 than to do 2 poseidon hashes. 

Definitely okay to keep this as is now since our shielded structure is small here (vs in network states i believe it's 8 elements). Wanted to note here for future reference!

File: client/server/silicon/silicon_utils.ts
Comment (lyronctk) [inline client/server/silicon/silicon_utils.ts:5]: BlindedMarket may be more appropriate since vanilla poseidon isn't technically an encryption algorithm (encryption implies a two way function)

It's a nit, but important since we're a cryptography focused company :)

File: client/server/silicon/silicon_utils.ts
Comment (lyronctk) [inline client/server/silicon/silicon_utils.ts:14]: May need to add a blinding factor here

Or else hash doesn't have 256 bits of security

Comment (lyronctk) [reply client/server/silicon/silicon_utils.ts:14]: your hash*

File: client/server/server.ts
Comment (lyronctk) [inline client/server/server.ts:9]: Are you keeping both markets up here for dev purposes?

**Comment by lyronctk** [comment]
Oh and more descriptive PR titles would be appreciated :)

e.g.
https://github.com/SeismicSystems/network-states-enclave/pull/17

File: client/server/silicon/silicon_utils.ts
Comment (lyronctk) [reply client/server/silicon/silicon_utils.ts:14]: although unsure where to best fit in the blinding factor with your setup. it could be that you're only using this to test and no problem

Comment (sfyll) [reply client/server/silicon/silicon_utils.ts:14]: this is replicating first and sole hash done on-chain when we init the "blinded markets", can go over in call if in dm if you want... so initial "blinding" is mostly cosmetics security wise.

File: client/server/server.ts
Comment (sfyll) [reply client/server/server.ts:9]: yes, haven't wrapped my head around that bit yet, so left it there... if you're seeing something I don't yet, let , me know !

File: client/server/silicon/silicon.ts
Comment (lyronctk) [reply client/server/silicon/silicon.ts:22]: still missing all the comments :)

File: client/server/silicon/silicon_utils.ts
Comment (lyronctk) [reply client/server/silicon/silicon_utils.ts:14]: understood!

**Review by lyronctk — APPROVED** [review]
approving now, but please add the function comments before you merge

**Pull request merged**
